src/lib.rs

```diff
@@ -190,10 +191,28 @@ where
         let frames = self.output_frames_next();
         let channels = self.nbr_channels();
         let mut wave_out = Vec::with_capacity(channels);
-        for _ in 0..channels {
-            wave_out.push(Vec::with_capacity(frames));
+        for chan in 0..channels {
+            let chan_out = if active_channels_mask.map(|mask| mask[chan]).unwrap_or(true)
+                // An empty input buffer is implicitly interpreted as if the
```

Comment:
I recommend to remove such implicit special case handling from the API.

Reply:
Yeah, it's convenient but probably not a great idea to keep it. It's from before I added the active channels mask. Let's remove.

Reply:
Thanks for confirming my concerns.

It should be fixed in a separate PR, because multiple tests need to be updated or deleted.